File: Matlab/kolus_button_replay.m
function replaybutton_record_neur_mic(handles)
tag = guidata(handles.f_main);
   [FileName,PathName,FilterIndex] = uigetfile('.mat','replay' ,tag.folder);
    load([PathName FileName]);

t_replay = input('Give start and stop time [start:stop], or no entry is full file');
if isempty(t_replay)
  t_replay = [1 tag.Duration]
else
  t_replay = sscanf(t_replay, '%f:%f');
end

%for use
scans = scans + refresh_time*Fs_mic;
pause(refresh_time)
data_globe.mic(max(1, scans - Fs_mic*refresh_time+1):scans)
%%%%Finalized button_record goes here

File: Python/kolus_button_replay.py
import numpy as np
from scipy.io import loadmat
from tkinter import filedialog
from tkinter import Tk

def replaybutton_record_neur_mic(handles):  # We'll skip 'handles' for simplicity

    # Simulate 'guidata(handles.f_main)' 
    tag = {'folder': '/path/to/default/folder'}  # Replace with appropriate path

    # File selection
    root = Tk()  # Need a Tkinter object for the file dialog
    root.withdraw()  # Hide the main window
    file_path = filedialog.askopenfilename(
        initialdir=tag['folder'],
        title='replay',
        filetypes=[('.mat', '*.mat')]
    )
    root.destroy()  # Remove the Tkinter object

    # Load data
    data = loadmat(file_path)  

    # Time range input
    t_replay_str = input('Give start and stop time [start:stop], or no entry is full file: ')
    if not t_replay_str:
        t_replay = [1, data['tag']['Duration']]  # Assuming 'Duration' exists in data
    else:
        t_replay = [float(x) for x in t_replay_str.split(':')]

    # Simulating variables/parameters 
    refresh_time = 0.1  # Replace with actual value 
    Fs_mic = 44100  # Replace with actual sampling rate
    data_globe = {'mic': np.random.rand(100000)}  # Simulating data_globe

    # Calculate indices
    scans = scans + refresh_time * Fs_mic  # Assuming 'scans' is defined elsewhere 
    start_index = max(1, scans - Fs_mic * refresh_time + 1)
    end_index = scans

    # Access data segment
    mic_segment = data_globe['mic'][start_index:end_index]

    # Placeholder for the rest of button_record functionality
    print("Finalized button_record logic would go here, using mic_segment") 
